applications/embeddings-modal/main.py

```diff
@@ -0,0 +1,50 @@
+import torch
+from transformers import AutoModel, AutoTokenizer
+
+base_name = "BAAI/bge-base-en-v1.5"
+small_name = "BAAI/bge-small-en-v1.5"
+large_name = "BAAI/bge-large-en-v1.5"
+
+
+class EmbeddingModel:
+  def __init__(self, model_name):
+    # Tokenizers convert input text into a format machine learning models can understand
+    # Splits text into tokens, encodes, pads, and truncates them
+    self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    # The model processes the tokenized-text into vectors that represent
+    # semantic meanings, relationships, and context
+    self.model = AutoModel.from_pretrained(model_name)
+    self.model.eval()
+
+  def get_embedding(self, text):
```

Comment:
Also check out Justin's pr

Reply:
ooh gotcha. I'll use it as a reference in setting up modal and FastAPI, model the code to reflect OpenAI, and update the readme and stuff. I'll add the code to run benchmarks in a separate PR too

Reply:
nope, hows the progress with baseten?